src/main/java/com/ays/common/util/validation/PhoneNumberValidator.java

```diff
@@ -1,44 +1,50 @@
 package com.ays.common.util.validation;
 
 import com.ays.common.model.AysPhoneNumber;
+import com.google.i18n.phonenumbers.PhoneNumberUtil;
+import com.google.i18n.phonenumbers.Phonenumber;
 import jakarta.validation.ConstraintValidator;
 import jakarta.validation.ConstraintValidatorContext;
 
 /**
  * A custom validator implementation for the {@link PhoneNumber} annotation.
- * Validates whether the provided {@link AysPhoneNumber} object is a valid phone number or not based on the
- * country code and line number length.
+ * Validates whether the provided {@link AysPhoneNumber} object is a valid phone number or not based on E.164 international standard.
  */
 class PhoneNumberValidator implements ConstraintValidator<PhoneNumber, AysPhoneNumber> {
 
     /**
-     * Validates whether the provided {@link AysPhoneNumber} object is a valid phone number or not based on the
-     * country code and line number length.
+     * Validates an AysPhoneNumber object based on E.164 international standard.
      *
-     * @param phoneNumber the {@link AysPhoneNumber} object to be validated
-     * @param context     the context in which the constraint is evaluated
-     * @return true if the phone number is valid, false otherwise
+     * @param phoneNumber The AysPhoneNumber object to be validated.
+     * @param context     The context for validation.
+     * @return True if the number is valid, false otherwise.
      */
     @Override
     public boolean isValid(AysPhoneNumber phoneNumber, ConstraintValidatorContext context) {
         final String countryCode = phoneNumber.getCountryCode();
         final String lineNumber = phoneNumber.getLineNumber();
+        final String fullNumber = "+" + countryCode + lineNumber;
```

Comment:
Burada bütün ülkeler için `+` kullanıldığına emin miyiz? Çok doğru bir kullanım olmayabillir. Eğer buradaki kısımları handle edemeyeceksek FE tarafında buradaki değerlerin toplu bir şekilde gönderilmesini isteyelim. FE tarafında bununla ilgili kütüphaneler bulunuyordu, toplu olarak tüm ülkelerin alan kodlarını bize gönderebiliyordu. Bu konuyu farklı bir issue'da da düşünebiliriz fakat bu şekilde statik yaptığımızda generic bir yapı olmamış oluyor gibi geliyor.

Reply:
Buradaki '+' karakteri doğrudan ülke koduyla alakalı değil. Uluslararası arama yapmak için ülke kodunun önüne koyduğun bir sembol. Country code listelerinin hiçbirinde zaten ülke kodlarının önünde '+' bulunmuyor.